### PR TITLE
Replace plain exception code with ERROR

### PR DESCRIPTION
**Pull request by oflebbe**
First step to remove unnecessary exceptions from the code

### CODE REVIEW

File: msktconf.cpp [hunk 1]
```diff
@@ -142,12 +142,12 @@ void create_fake_krb5_conf(msktutil_flags *flags)
 #ifdef HAVE_SETENV
     int ret = setenv("KRB5_CONFIG", g_config_filename.c_str(), 1);
     if (ret) {
-        throw Exception("setenv failed");
+        ERROR("setenv failed");
     }
 #else
     int ret = putenv(strdup((std::string("KRB5_CONFIG=") +  g_config_filename).c_str()));
```
Comment (michael-o) [inline msktconf.cpp]: Wo does the  `free` after `strdup` because it uses `malloc` in the background?

**Comment by michael-o** [comment]
These exits happen before `g_context` has been initialized?

**Comment by oflebbe** [comment]
Updated the branch. Do not use an C preprocessor directive, rather a plain function. removed on duplicate error.

Updated again to remove spurious search and replace error

File: msktutil.h [hunk 2]
```diff
@@ -294,6 +294,8 @@ int generate_new_password(msktutil_flags *flags);
 /* Verbose messages */
 #define VERBOSE(text...) if (g_verbose) { fprintf(stdout, " -- %s: ", __FUNCTION__); fprintf(stdout, ## text); fprintf(stdout, "\n"); }
 
+/* Fatal error */
+void error_exit( const char *text);
```
Comment (michael-o) [inline msktutil.h]: Leading space.

File: msktutil.cpp [hunk 3]
```diff
@@ -41,6 +41,11 @@
 
 int g_verbose = 0;
 
+/* Fatal error */
+void error_exit( const char *text) {
+    fprintf(stderr, "error_exit: %s: %s\n", text, strerror(errno));
```
Comment (michael-o) [inline msktutil.cpp]: Does the user really case that `error_exit` is used here? I think all he needs to see is `ERROR`. Everything else is an implementation detail.

File: msktutil.cpp [hunk 4]
```diff
@@ -60,7 +65,7 @@ std::string sform(const char* format, ...)
     int result = vasprintf(&buf, format, args);
 #endif
     if (result < 0) {
-        throw Exception("vasprintf error");
+        error_exit("vasprintf error");
```
Comment (michael-o) [inline msktutil.cpp]: `error` not `failed`?

File: msktutil.cpp (hunk 3 above)
Comment (oflebbe) [inline msktutil.cpp]: The user may not care what failed. But it helps us to diagnose failures. BTW, the message may be seen even before, but was implementation dependent, AFAIK

File: msktutil.cpp (hunk 4 above)
Comment (oflebbe) [inline msktutil.cpp]: done

**Comment by oflebbe** [comment]
@michael-o regarding "These exits happen before g_context has been initialized?" 
Why does this matter ?

**Comment by michael-o** [comment]
@oflebbe `initialize_g_context()` is called as second statement in `main`. Only those changed spots are invoked. Error conditions call `exit` and `destroy_g_context` is never called. In fact, it is not destroyed in current master either. Might this be an issue? It is at least not clean code. Am I wrong?

**Comment by oflebbe** [comment]
A kerberos context is an in-memory data structure. Since we invoke error_exit when a fatal error occured, we do not care about process memory of a process going to die instantly. 

Cleaning up the kerberos context will not have any effect to the system at this point in time.

**Comment by oflebbe** [comment]
I am not saying you are wrong, but the programm is doing a "Whoaa something terrible happend!" with no option to continue sanely. A clean error message is the way to end. 

If we would have to design a library function, this would be a no-go, since a lib should never call exit() and never eat memory, silently.

**Comment by oflebbe** [comment]
Hey this was my first github merge ;-)

**Comment by michael-o** [comment]
You should have squashed them first and rebased on top of master. The log doesn't look appealing to the reader now.

**Pull request merged**